mlflow/utils/gorilla.py

```diff
@@ -331,6 +334,41 @@ def apply(patch):
     setattr(patch.destination, _ACTIVE_PATCH, patch)
 
 
+def revert(patch):
+    """Revert a patch.
+    Parameters
+    ----------
+    patch : gorilla.Patch
+        Patch.
+    Note
+    ----
+    This is only possible if the attribute :attr:`Settings.store_hit` was set
+    to ``True`` when applying the patch and overriding an existing attribute.
+    """
+    if getattr(patch.destination, _ACTIVE_PATCH, None) != patch:
+        return
+
+    try:
+        original = get_original_attribute(patch.destination, patch.name)
+    except AttributeError:
+        raise RuntimeError(
+            "Cannot revert the attribute named '%s' since the setting "
+            "'store_hit' was not set to True when applying the patch."
+            % (patch.destination.__name__,)
+        )
+
+    original_name = _ORIGINAL_NAME % (patch.name,)
+    if not getattr(patch.destination, original_name):
+        return
+
+    setattr(patch.destination, patch.name, original)
+
+    if original_name in patch.destination.__dict__:
```

Comment:
Only delete added attributes by gorilla monkey-patching, if attribute is defined on the class as opposed to inherited from a parent class.

Reply:
Awesome solution! In accordance with that StackOverflow comment that we discussed offline, can we also add a `hasattr` check here too?

Reply:
For class attributes stored in [slots](https://stackoverflow.com/questions/472000/usage-of-slots), if no additional attrbiutes are added then there will be no __dict__. Any [monkey patching will add attributes ](https://github.com/mlflow/mlflow/blob/de01ccf24474d8f9479851838292caa0d62865ec/mlflow/utils/gorilla.py#L334)forcing the class to have a __dict__ field on the class. If no code was mokey patched and the slight possibility of class to only have attributes defined via slots, i.e. __dict__  is undefined then the [getattr](https://github.com/mlflow/mlflow/blob/de01ccf24474d8f9479851838292caa0d62865ec/mlflow/utils/gorilla.py#L348) in the beginning of the method will short circuit. 

TLDR: No need to add  hasattr check for slots edge case.